src/MICore/Debugger.cs

```diff
@@ -482,10 +497,17 @@ public Task CmdStart()
             return CmdAsync("-exec-run", ResultClass.running);
         }
 
-        protected bool _requestingRealAsyncBreak = false;
-        public Task CmdBreak()
+        public enum BreakRequest
+        {
+            None,
+            Async,
+            Stop,
+            Internal
+        }
+        protected BreakRequest _requestingRealAsyncBreak = BreakRequest.None;
+        public Task CmdBreak(BreakRequest request)
         {
-            _requestingRealAsyncBreak = true;
+            _requestingRealAsyncBreak = request;
```

Comment:
Did you forget to add your changes? This is still just overwriting the variable.